sktime/param_est/plugin/_transformer.py

```diff
@@ -0,0 +1,240 @@
+
+    >>> # trafo is a forecaster with a "sp" parameter which we want to tune
+    >>> trafo = Deseasonalizer()  # doctest: +SKIP
+    >>> sp_auto = PluginParamsTransformer(sp_est, trafo)  # doctest: +SKIP
+    >>>
+    >>> # fit sp_auto to data, transform, and inspect the tuned sp parameter
+    >>> sp_auto.fit(X)  # doctest: +SKIP
+    PluginParamsTransformer(...)
+    >>> Xt = sp_auto.transform(X)  # doctest: +SKIP
+    >>> sp_auto.transformer_.get_params()["sp"]  # doctest: +SKIP
+    12
+    >>> # shorthand ways to specify sp_auto, via dunder, does the same
+    >>> sp_auto = sp_est * trafo  # doctest: +SKIP
+    >>> # or entire pipeline in one go
+    >>> sp_auto = Differencer() * SeasonalityACF() * Deseasonalizer()  # doctest: +SKIP
+
+    using dictionary to plug "foo" parameter into "sp"
+
+    >>> from sktime.param_est.fixed import FixedParams
+    >>> sp_plugin = PluginParamsTransformer(
+    ...     FixedParams({"foo": 12}), Deseasonalizer(), params={"sp": "foo"}
+    ... )  # doctest: +SKIP
+    """
+
+    _tags = {
+        "capability:inverse_transform": True,  # can the transformer inverse transform?
+        "univariate-only": False,  # can the transformer handle multivariate X?
+        "fit_is_empty": False,  # is fit empty and can be skipped? Yes = True
+        "remember_data": False,  # whether all data seen is remembered as self._X
+    }
+
+    # attribute for _DelegatedTransformer, which then delegates
+    #     all non-overridden methods to those of same name in self.transformer_
+    #     see further details in _DelegatedTransformer docstring
+    _delegate_name = "transformer_"
+
+    def __init__(self, param_est, transformer, params=None, update_params=False):
+        self.param_est = param_est
+        self.param_est_ = param_est.clone()
+        self.transformer = transformer
+        self.transformer_ = transformer.clone()
+        self.params = params
+        self.update_params = update_params
+
+        super().__init__()
+
+        TAGS_TO_CLONE = [
+            "scitype:transform-input",
+            "scitype:transform-output",
+            "scitype:transform-labels",
+            "scitype:instancewise",
+            "capability:inverse_transform",
+            "capability:inverse_transform:range",
+            "capability:inverse_transform:exact",
+            "univariate-only",
+            "y_inner_mtype",
+            "requires_y",
+            "enforce_index_type",
+            "X-y-must-have-same-index",
+            "transform-returns-same-time-index",
+            "skip-inverse-transform",
+            "capability:unequal_length",
+            "capability:unequal_length:removes",
+            "handles-missing-data",
+            "capability:missing_values:removes",
+        ]
+
+        self.clone_tags(self.transformer_, TAGS_TO_CLONE)
+
+        # todo: only works for single series now
+        #   think about how to deal with vectorization later
```

Comment:
Does it mean as of now it can't fit independently for each of the columns in a dataframe? What happens instead?

Reply:
you can, with a `ColumnForecaster`

Reply:
or, what should happen in your opinion, based on different conditions?